internal/common/utils.go

```diff
@@ -50,30 +50,29 @@ func CommandValueToReading(cv *dsModels.CommandValue, devName string, encoding s
 }
 
 func SendEvent(event *dsModels.Event) {
-	ctx := context.WithValue(context.Background(), CorrelationHeader, uuid.New().String())
-	ct := clients.ContentTypeJSON
+	correlation := uuid.New().String()
+	ctx := context.WithValue(context.Background(), CorrelationHeader, correlation)
 	if event.HasBinaryValue() {
-		ct = clients.ContentTypeCBOR
-		if len(event.EncodedEvent) <= 0 {
-			var err error
-			event.EncodedEvent, err = event.EncodeBinaryEvent(&event.Event)
-			if err != nil {
-				LoggingClient.Error("ERROR encoding binary event!")
-			}
-			LoggingClient.Info(fmt.Sprintf("EncodedEvent within SendEvent: %v", string(event.EncodedEvent[:20])))
+		ctx = context.WithValue(ctx, clients.ContentType, clients.ContentTypeCBOR)
+	} else {
+		ctx = context.WithValue(ctx, clients.ContentType, clients.ContentTypeJSON)
+	}
+	// Call MarshalEvent to encode as byte array whether event contains binary or JSON readings
+	var err error
+	if len(event.EncodedEvent) <= 0 {
+		event.EncodedEvent, err = EventClient.MarshalEvent(event.Event)
+		if err != nil {
+			LoggingClient.Error(fmt.Sprintf("SendEvent [correlationid %s] Error encoding event for device %s: %v", correlation, event.Device, err))
 		} else {
-			// using existing CBOR encoded event to send over to CoreData...
-			LoggingClient.Info(fmt.Sprintf("EncodedEvent already prepared: %v", string(event.EncodedEvent[:20])))
+			LoggingClient.Info(fmt.Sprintf("SendEvent [correlationid %s] EventClient.MarshalEvent encoded event: %.200v", correlation, event))
```

Comment:
Ah these could be adjusted to comply with structured logging. k
It currently logs the first 200 chars of an event. This was sufficient to show full events for SwitchButton (since it sends very simple JSON payload) and the first portion of binary image data when we send the Image payload. I found this improved the DX OOBE since it allowed running device-simple example with a straightforward correlation between a device's AutoEvents and activities/events received by Core services.
If INFO log level is anticipated for use in production then yes these should be debug or trace (trace if we log entire event instead of just first 200 chars). I reverted that change actually in order to support straightforward integration testing.
I saw a lot of 'noise' when setting the simple example to use debug log level though.

Reply:
Trace is for when we log correlation id...

Reply:
@lenny-intel  I made it consistent with edgex-core-data (Info level) with correlation id but no event data; I added contentType to the structured log entry for convenience. I used Trace level logs with correlation id and event data just when it is actually sent. These changes are in PR #293 though (I added you to that review).